src/NexusMods.App.UI/IImageStore.cs

```diff
@@ -0,0 +1,16 @@
+using Avalonia.Media.Imaging;
+using JetBrains.Annotations;
+using NexusMods.MnemonicDB.Abstractions;
+
+namespace NexusMods.App.UI;
+
+/// <summary>
+/// Optimized storage for images.
+/// </summary>
+[PublicAPI]
+public interface IImageStore
+{
+    ValueTask Store(EntityId id, Bitmap bitmap);
+
+    ValueTask<Bitmap?> Retrieve(EntityId id);
```

Comment:
Should we use Nominal Typing and make subclass for `EntityId` for this?

Reply:
Ok I'm reading a bit more of the code and I see that Entity is just a unique identifier. This isn't using MnemonicDB in any way, it just needs a unique id.

Could you add some comments on Store and Retrieve to clarify this?